sdk/communication/azure-communication-chat/src/main/java/com/azure/android/communication/chat/ChatThreadAsyncClient.java

```diff
@@ -223,11 +225,35 @@ public String getChatThreadId() {
     @ServiceMethod(returns = ReturnType.SINGLE)
     public CompletableFuture<Response<AddChatParticipantsResult>> addParticipantWithResponse(
         ChatParticipant participant, Context context) {
-        return this.addParticipants(
+        return this.addParticipant(
             new AddChatParticipantsOptions().setParticipants(Collections.singletonList(participant)),
             context);
     }
 
+    /**
+     * Adds a participant to a thread. If participants already exist, no change occurs.
+     *
+     * @param options Options for adding participants.
+     * @param context The context to associate with this operation.
+     * @throws InvalidParticipantException thrown if the participant is rejected by the server.
+     *
+     * @return the {@link CompletableFuture} that emits response containing the operation result.
+     */
+    CompletableFuture<Response<AddChatParticipantsResult>> addParticipant(AddChatParticipantsOptions options,
+                                                                           Context context) {
+        context = context == null ? Context.NONE : context;
+        return this.addParticipants(options, context)
+            .thenApply(result -> {
+                if (result.getValue().getInvalidParticipants() != null) {
+                    if (result.getValue().getInvalidParticipants().size() > 0) {
+                        ChatError error = result.getValue().getInvalidParticipants().get(0);
+                        this.logger.logExceptionAsError(new InvalidParticipantException(error));
```

Comment:
To be honest, I don't have much experience on Android dev. I am not sure whether this is the proper way for throwing a exception. Do you have any suggestions?

Reply:
The return type composes the error field and has no other fields. If the expectation is to throw this composed error from here, then should we make the return type Void.

Reply:
I see this is a common internal function used from two other places
1. `public CompletableFuture<Void> addParticipant(..)`
2. `public CompletableFuture<Response<AddChatParticipantsResult>> addParticipantWithResponse(..)`

Are you expecting not to throw from 2 but only from 1, then you you might want to do the check and throw only from 1.

If you expect to throw an exception from both 1 and 2, you might want to change the return type of 2 to `CF<Response<Void>>` and add check and throw.

Reply:
Correct. The expectation is to throw for both 1 and 2. I'll fix it.